File: 백준/Gold/2023. 신기한 소수/신기한 소수.py
n = int(input())


# 소수 판별
def isPrime(a):
    if (a < 2):
        return False
    for i in range(2, int(a ** 0.5) + 1):
        if (a % i == 0):
            return False
    return True


def dfs(num):
    # 목표 길이 도달 시 멈춤
    if len(str(num)) == n:
        print(num)
    else:
        for i in range(10):
            temp = num * 10 + i
            if isPrime(temp):
                dfs(temp)


dfs(2)
dfs(3)
dfs(5)
dfs(7)

File: 백준/Gold/2023. 신기한 소수/신기한 소수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;

public class Main {
    static int N;
    static int[] list = {2, 3, 5, 7};
    static ArrayList<Integer> arr = new ArrayList<>();
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        Boolean[] visited = new Boolean[N];
        for (Integer i : list) {
            dfs(i);
        }
    }

    static void dfs(Integer i){
        String s = String.valueOf(i);

        if(s.length() == N){
            System.out.println(s);
            return;
        }

        for (int j=0; j<10; j++) {
            Integer value = i * 10 + j;
            if(isPrime(value)){
                dfs(value);
            }

        }
    }

    static Boolean isPrime (Integer value){
        for(int i=2; i<value/2 + 1; i++){
            if(value % i == 0){
                return false;
            }
        }
        return true;
    }
}
